pyked/validation.py

```diff
@@ -252,14 +252,20 @@ def _validate_isvalid_history(self, isvalid_history, field, value):
                         'with ' + property_units['time'])
 
         # Check that the values have the right number of columns
-        n_cols = len(value['values'][0])
-        max_cols = max(value['time']['column'],
-                       value['quantity']['column'],
-                       value.get('uncertainty', {}).get('column', 0)) + 1
-        if n_cols > max_cols:
-            self._error(field, 'too many columns in the values')
-        elif n_cols < max_cols:
-            self._error(field, 'not enough columns in the values')
+        # If reading from a file, the file will not be validated.
+        # A file can have an arbitrary number of columns, and the columns
+        # to be used are specified.
+        if type(value['values']) is list:
```

Comment:
Why not `'filename' not in value['values'].keys()`?

Reply:
I agree that we can assume the CSV file is specified relative to the yaml file, but `DataPoint` doesn't know where the yaml file directory is unless it's specified as an argument to `__init__`, right? Is there a simpler way to deal with a relative path?

I tried using `'filename' not in value['values'].keys()` in a [previous commit](https://github.com/pr-omethe-us/PyKED/pull/115/commits/abe35c49c489450d088d4c0c9ee37dd1235daa21) but this fails when the values are given as a list. In that case, `value['values']` is a list, not a dict, so `if 'filename' not in value['values'].keys()` raises an error.

Reply:
As far as I know, just trying to open the file will try to open it relative to the working directory of the Python process. Are you saying that if I have a file structure like
```
|- database
|---butanol
|------file_1.yaml
|------file_1.csv
```

and I start Python in the `database` directory, and load `file_1.yaml` like
```
>>> ChemKED('butanol/file_1.yaml')
```
it won't work, because Python will assume the `file_1.csv` is relative to `database`, not `butanol`?

As on your other PR, I think the simpler/more "pythonic" way to do this is a `try...except` block, rather than checking the type of the value.

Reply:
Yes, that's exactly what I'm saying. I was using a script to read multiple yaml files in a complex directory structure within a `database` directory, and python was looking for the csv file in `database`, not in the folder with the yaml file.

I'm not sure that a `try...except` block would work well here. Are you thinking of something like? It seems more complex and confusing this way, and it puts a lot of code between the `try` and `except`
```
try:
    if 'filename' in value['values'].keys():
        # Don't do anything because csv files aren't validated
        pass
    else:
        # This should never happen. If vale['values'] is a dictionary with keys, 'filename' should be a key
        self._error(field, 'must include filename or list of values')
except KeyError:
        # value['values'] is probably a list.
        # Code from earlier that checks the number of columns
```

Since this PR is related to time histories, I have a somewhat related question for you that I just stumbled on. Let's say somebody has a csv file with three columns - time, pressure, and volume. Right now, these must be implemented as two separate `time-histories` and the csv will be loaded twice, right? Is there interest in allowing the user to specify multiple time-histories using a single file?

Reply:
Just to keep this focused on the code here, I moved the other discussion to the main comment thread. Anyhow, two things:
1) I want to validate csv files in the future, so we might as well set up for that here
2) The code could be

```
try:
    if 'filename' not in value['values'].keys():
        self._error(field, 'must include filename or list of values')
except TypeError:
    # Code from earlier that checks the number of columns
```
or
```
try:
    filename = value['values'].get('filename')
    if filename is None:
        self._error(field...
except AttributeError:
    # Code from earlier
```
which isn't all that confusing to me. The reason (to me) to avoid the `type` function is that it doesn't always handle inheritance in a straightforward way, so we'd be relying on the underlying YAML library to always return something that's a subclass of list. On the other hand, with the try-except, we're using the duck-typing in Python to try something that we expect to be the case, and catch the resulting errors.